test/connection_test.py

```diff
@@ -0,0 +1,138 @@
+# Databricks notebook source
+from data_io.connection import build_connection
+from urlpath import URL
+from pyspark.sql import SparkSession
+from pyspark.dbutils import DBUtils
+"""
+this class is for test mining the credentials
+"""
+
+class GetConnection():
+  
+  def __init__(self):
+    spark = SparkSession.builder.getOrCreate()
+    dbutils = DBUtils(spark)
+    self.scope = (
+                dbutils
+                .notebook
+                .entry_point
+                .getDbutils()
+                .notebook()
+                .getContext()
+                .tags()
+                .apply('user')
+                # XXX Hack incoming
+                .split('@')
+                [0]
+            )
+  # get secret
+  def get_secret(self, key):
+    spark = SparkSession.builder.getOrCreate()
+    dbutils = DBUtils(spark)
+    return dbutils.secrets.get(scope=self.scope, key=key)
+    
+  # providing list of creds
+  def get_connections(self):
+    list_creds = []
+    secret = self.get_secret('credentials.cfg')
+    connection_list = []
+    for st in secret.split('\n'):
+      # checking if this is connection
+      try:
+        if('[' == st[0]):
+          # appending connection
+          connection_list.append(st.split('[')[1].split(']')[0])
+      except:
+       # string has no lengh
+       print('no index')
+    # getting each url in credential file
+    list_cred = []
```

Comment:
Please use standard parsers (see above).

Reply:
same as above the parser was not working in the unittest. see [example](https://tfs-edp.cloud.databricks.com/?o=2841493856018718#notebook/4029213076043214/command/4029213076043215) of intially figuring out how to build a unittest on command 18

Reply:
I'm not understanding what is not working.